### ISSUE TITLE
Could be possible to switch to gix?

### ISSUE THREAD

**Issue by lu-zero** [lukaslueg/built#66]
It would be less problematic than relying on the presence of libgit2.

I can contribute the code to switch to it if you deem it acceptable.

**Comment by lukaslueg** [comment]
That's certainly possible, although reliability / platform compatibility would be paramount.

There has been ongoing [work](https://github.com/lukaslueg/built/pull/52) ([and](https://github.com/Byron/built/pull/1)) to add the required features to gitoxide. If you ping @Byron, he may be able to guide towards what's left to do.

**Comment by lu-zero** [comment]
I see he started working on a new branch. I forgot the dirty check is missing :)

**Comment by Byron** [comment]
Indeed, there really is 'just' [this PR](https://github.com/Byron/gitoxide/pull/1368) missing which I plan to work on for months now. It will happen though…I am just spread thin ATM.

**Comment by jayvdb** [comment]
https://github.com/Byron/gitoxide/pull/1368 was merged in June. Maybe this is now feasible?

**Comment by Byron** [comment]
Please note that I have been using these PRs for other things and keep reopening them, so the mere merge doesn't really say much about its contents.
Thus, `is_dirty` isn't fully supported yet as it also needs a diff between a `HEAD` and the index, which isn't implemented yet.
And as I am tracking this and have it on my radar, you can trust I will be back here to finish the integration as soon as it's possible.

Let me finish with my sincere apologies for the noise.

**Comment by artob** [comment]
This would be great, as this crate pulling in libgit2 breaks e.g. Android builds several levels up the dependency tree.

**Comment by lukaslueg** [comment]
@artob, as far as I knew libgit2 itself does build on Android, at least in theory. Would you mind a quick note on what causes the breakage?

**Comment by lu-zero** [comment]
I guess it could be a fun task to benchmark the different opensource models since it is well contained. Worst case we have a nice draft that can be hand-polished.

**Comment by Byron** [comment]
Great! I had a manual implementation already, but just started another automated one to see how it goes: https://github.com/Byron/built/pulls .

**Comment by Byron** [comment]
I think you have to tell it to keep `git2` in tests. That's what I did for the manual port. Alternatively, the test-suite has to be changed to `gix-testtools` with shell scripts for initial states, which would work as well.
